migrator/migrations/m_196_to_m_197_set_poduid_where_null/migration_impl.go

```diff
@@ -0,0 +1,241 @@
+	plops := make([]*storage.ProcessListeningOnPortStorage, batchSize)
+	count := 0
+	err := plopStore.Walk(ctx,
+		func(plop *storage.ProcessListeningOnPortStorage) error {
+			// Don't set the PodUid if it is already set. The process information has the
+			// podid, so it must be set to proceed.
+			if plop.GetPodUid() == "" && plop.GetProcess() != nil {
+				podKey := getPodKey(plop.GetProcess().GetPodId(), plop.GetDeploymentId())
+				podUID, exists := podUIDMap[podKey]
+				if exists {
+					plop.PodUid = podUID
+					plops[count] = plop
+					count++
+				}
+
+				if count == batchSize {
+					err := plopStore.UpsertMany(ctx, plops)
+					count = 0
+					if err != nil {
+						return err
+					}
+				}
+			}
+
+			return nil
+		})
+
+	if count > 0 {
+		plops = plops[:count]
+		err := plopStore.UpsertMany(ctx, plops)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+// Sets PodUids by using information from the pods table. It first walks through the pods table and
+// makes a table where the key is a combination of the pod name and deploymentid and the value is the
+// PodUid. It then walks through the listening_endpoints table and uses the map to set the value of
+// PodUid where it is null and the process information exists. This is effectively a join. If the
+// process information does not exist the podid (pod name) doesn't exist, and the PodUid cannot be set
+// using this strategy.
+func setPodUIDsUsingPods(ctx context.Context, podStore podDatastore.Store, plopStore plopDatastore.Store, batchSize int) error {
+	podUIDMap, err := getPodUIDMap(ctx, podStore)
+	if err != nil {
+		return err
+	}
+
+	err = setPodUidsUsingPods(ctx, plopStore, podUIDMap, batchSize)
+
+	return err
+}
+
+// There are two strategies for setting PodUid in the listening_endpoints table. The first is to use
+// the process_indicators table. The listening_endpoints table can be joined to the process_indicators
+// table using the processindicatorid column from listening_endpoints and the id column from the
+// process_indicators table. Then the PodUid can be set using the value from the process_indicators
+// table. In cases where the processindicatorid does not have any matching process indicator the pods
+// table can be used to set the PodUid. In cases where there is a matching process indicator the
+// listening_endpoints table stores no process information and in cases where there is no matching
+// process indicator the listening_endpoints tables store process information, including the podid.
+// The podid is not the PodUid. It is the name of the pod. The podid along with the deploymentid
+// can be used to do a join on the pods table and obtain the PodUid. Where one strategy doesn't work
+// the other should work, unless the pod for the listening endpoint has been deleted.
+func updateGlobalScope(ctx context.Context, database *types.Databases) error {
+	batchSize := 2000
```

Comment:
I set batchSize here instead of making it a global, because I wanted to use other values of batchSize in the unit tests.